src/multi.c

```diff
@@ -69,6 +70,7 @@ void resetClientMultiState(client *c) {
     c->mstate->cmd_inv_flags = 0;
     c->mstate->argv_len_sums = 0;
     c->mstate->alloc_count = 0;
+    c->mstate->transaction_db_id = c->db->id;
```

Comment:
Why initialize it while cleaning up. I think it should suffice if we should set the value while the client is entering into multi state and update it within the multi flow if things change.